pkg/endpoint/policy.go

```diff
@@ -731,11 +731,19 @@ func (e *Endpoint) regenerate(owner Owner, reason string) (retErr error) {
 	}()
 
 	revision, compilationExecuted, err = e.regenerateBPF(owner, tmpDir, reason)
+	if err != nil {
+		failDir := e.FailedDirectoryPath()
+		scopedLog.WithFields(logrus.Fields{
+			logfields.Path: failDir,
+		}).Warn("generating BPF for endpoint failed, keeping stale directory.")
+		os.Rename(tmpDir, failDir)
+		return err
+	}
 
 	// Depending upon result of BPF regeneration (compilation executed, or
```

Comment:
Update comment to reflect removal of error parameter.

Reply:
Done